core/src/datayoga_core/utils.py

```diff
@@ -108,3 +110,23 @@ def add_uid(record: Dict[str, Any]) -> Dict[str, Any]:
 
 def remove_msg_id(record: dict) -> dict:
     return {k: v for k, v in record.items() if k != Block.MSG_ID_FIELD}
+
+
+def explode_records(records: List[Dict[str, Any]], field_expression: str) -> List[Dict[str, Any]]:
+    field_name, expression = map(str.strip, field_expression.split(":", maxsplit=1))
+
+    jmespath_expr = JMESPathExpression()
+    jmespath_expr.compile(expression)
+
+    exploded_records = []
+
+    for record in records:
+        exploded_fields = jmespath_expr.search(record)
+
+        if exploded_fields:
+            for exploded_field in exploded_fields:
```

Comment:
Consider changing the name - exploded_fields implies it is multiple fields, but really this holds the array of one field in multiple records

Reply:
right, changed.